### ISSUE TITLE
Exporting and using bezier curves

### ISSUE THREAD

**Issue by felipetavares** [GoranM/bdx#659]
Bezier curves are very useful in game development to pre-determine paths objects should follow.

Currently BDX has no support to exporting curves from blender and no support for them at runtime.

The idea here is to extend the exporter to add a special "curve" property to objects of the type "CURVE". This property would be an array containing 3d arrays representing each point of the bezier curve in the sequence: knot 1, handle 1, handle 2, knot2, knot3, handle 3 etcetera

This would look something like:

```
{
...
"curve": [
   [-1.0, 0.0, 0.0],
   [-0.5, 0.5, 0.0],
   [-9.15615463256836, 8.542720794677734, -8.098329544067383],
   [-8.15615463256836, 8.542720794677734, -8.098329544067383],
   ...],
...
}
```

At runtime, objects of the type "CURVE" would be instantiated from a Curve class (that extends GameObject). This class would provide at least a method `Vector3f get(float t)` that would allow the end user to get the 3d position of the curve at linear position `t`.

I already have most of the implementation for this done, but before submitting a PR I think it's better to check if there aren't better ideas.

**Comment by SolarLune** [comment]
Yo! I agree curves are useful, and this sounds like a logical enough addition. Curves can be useful for moving platforms, for NPC paths, for all kinds of things.

Another function that would be nice would be size(), which returns the number of points in the path. Maybe a switch to turn it from a bezier curve to a linear path, as well.

**Comment by GoranM** [comment]
Seems like a reasonable approach. Feel free to make the PR when ready.

**Comment by felipetavares** [comment]
*libgdx* seems to have some relevant classes:

* [Bezier][0]
* [Path][1]
* [BSpline][2]

[0]: https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/Bezier.html
[1]: https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/Path.html 
[2]: https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/BSpline.html

I'll check if our implementation can take advantage of them (it probably can).

**Comment by SolarLune** [comment]
After @felipetavares looks into this (whether implementable or not), could we get a new release, @GoranM?

**Comment by GoranM** [comment]
Sure.

**Comment by dualjack** [comment]
" I'm so excited! "

[](https://youtu.be/8iwBM_YB1sE?t=58s)

**Comment by SolarLune** [comment]
@felipetavares - Did you happen to find anything about this?

**Comment by felipetavares** [comment]
@SolarLune Not really, got dragged into work (still am). If someone has interest in going forward with this, I can send my current code.

**Comment by SolarLune** [comment]
That sounds fine. If you put it up in a fork of BDX here on Github, then it can be available to peruse for whomever wants to implement it (though I might take a crack at it).
